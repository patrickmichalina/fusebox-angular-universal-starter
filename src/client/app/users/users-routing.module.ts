import { UsersComponent } from './users.component'
import { NgModule } from '@angular/core'
import { RouterModule } from '@angular/router'
import { MetaGuard } from '@ngx-meta/core'
import { AdminGuard } from '../shared/services/guard-admin.service'

@NgModule({
  imports: [
    RouterModule.forChild([
      {
        path: '',
        component: UsersComponent,
        canActivate: [MetaGuard, AdminGuard],
        data: {
          meta: {
            title: 'i18n.users.title',
            description: 'i18n.users.description'
          },
          response: {
            cache: {
              directive: 'private'
            }
          }
        }
      }
    ])
  ],
  exports: [RouterModule]
})
export class UsersRoutingModule { }
