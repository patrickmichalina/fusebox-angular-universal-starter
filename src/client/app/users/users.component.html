<div fxFlexFill fxLayout="column">
  <mat-nav-list fxFlex>
    <mat-list-item *ngFor="let user of users$ | async; trackBy: trackByUsers">
      <img matListAvatar [src]="user.photo">
      <h4 mat-line>{{ user.displayName }}</h4>
      <p mat-line> {{ user.email }} </p>
    </mat-list-item>
  </mat-nav-list>
  <!-- <mat-paginator (page)="pageEvent($event)" [length]="100" [pageSize]="10" [pageSizeOptions]="[5, 10, 25, 100]"></mat-paginator> -->
</div>