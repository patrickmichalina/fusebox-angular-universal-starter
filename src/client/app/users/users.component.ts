import { Observable } from 'rxjs/Observable'
import { ChangeDetectionStrategy, Component } from '@angular/core'
import { FirebaseDatabaseService } from '../shared/services/firebase-database.service'
import { SettingService } from '../shared/services/setting.service'
// import { PageEvent } from '@angular/material'

@Component({
  selector: 'pm-users',
  templateUrl: './users.component.html',
  styleUrls: ['./users.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class UsersComponent {

  public usersFromDb$ = this.db
    .getListKeyed('users', ref => ref.limitToFirst(5000)) // TODO: pagination

  public users$ = Observable.combineLatest(this.usersFromDb$, this.ss.settings$,
    (userInfo, settings) => {
      return userInfo.map((user: any) => {
        return {
          ...user,
          photo: user.photo || settings.assets.userAvatarImage
        }
      })
    })

  constructor(private db: FirebaseDatabaseService, private ss: SettingService) { }

  // pageEvent(pageEvent: PageEvent) {
  //   console.log(pageEvent)
  // }

  trackByUsers(index: number, item: any) {
    return item.id
  }
}
