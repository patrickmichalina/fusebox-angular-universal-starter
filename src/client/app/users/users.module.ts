import { UsersRoutingModule } from './users-routing.module'
import { UsersComponent } from './users.component'
import { NgModule } from '@angular/core'
import { SharedModule } from '../shared/shared.module'

@NgModule({
  imports: [UsersRoutingModule, SharedModule],
  declarations: [UsersComponent],
  exports: [UsersComponent]
})
export class UsersModule { }
