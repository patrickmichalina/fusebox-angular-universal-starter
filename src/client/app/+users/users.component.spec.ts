import { UsersComponent } from './users.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'
import { UsersModule } from './users.module'

@Component({
  selector: 'test-component',
  template: '<pm-users></pm-users>'
})
class TestComponent {}

describe(UsersComponent.name, () => {
  let fixture: ComponentFixture<TestComponent>

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [UsersModule],
      declarations: [TestComponent]
    }).compileComponents()
  }))

  beforeEach(async(() => {
    fixture = TestBed.createComponent(TestComponent)
  }))

  afterEach(async(() => {
    TestBed.resetTestingModule()
  }))

  it('should compile', async(() => {
    expect(fixture.nativeElement).toBeDefined()
    expect(fixture).toMatchSnapshot()
  }))
})
