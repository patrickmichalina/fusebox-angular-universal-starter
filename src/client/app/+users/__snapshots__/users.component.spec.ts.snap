// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersComponent should compile 1`] = `
<test-component>
  <pm-users>
    <div
      fxflexfill=""
      fxlayout="column"
      style="height: 100%; margin: 0px; min-height: 100%; min-width: 100%; width: 100%;"
    >
      
    
      <mat-nav-list
        class="mat-nav-list"
        fxflex=""
        role="list"
      >
        
      
        
        
    
      </mat-nav-list>
      
    
      
  
    </div>
  </pm-users>
</test-component>
`;
